src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinRTWebSocket.cs

```diff
@@ -515,5 +539,51 @@ private void UpdateStateCloseReceived()
             }
         }
         #endregion
+
+        #region Helpers
+        private static bool InitMessageWebSocketClientCertificateSupported()
+        {
+            return ApiInformation.IsPropertyPresent(
+                "Windows.Networking.Sockets.MessageWebSocketControl",
+                "ClientCertificate");
+        }
+
+        // There are currently only two ways to convert a .NET X509Certificate object into a WinRT Certificate without
```

Comment:
Ok. In reality X509Certificate2 is a derived class of X509Certificate. So, any X509Certificate2 is already an X509Certificate object.

As long as you don't need to call X509Certificate2 methods directly, it doesn't matter. If you did the benefit of calling the extra X509Certificate2 methods, you could test the up-casting to X509Certificate2.